File: SDKs/amazonaws.com/opsworks/2013-02-18/go/pkg/models/operations/registerinstance.go
package operations

import (
	"openapi/pkg/models/shared"
)

type RegisterInstanceXAmzTargetEnum string

const (
	RegisterInstanceXAmzTargetEnumOpsWorks20130218RegisterInstance RegisterInstanceXAmzTargetEnum = "OpsWorks_20130218.RegisterInstance"
)

type RegisterInstanceHeaders struct {
	XAmzAlgorithm     *string                        `header:"style=simple,explode=false,name=X-Amz-Algorithm"`
	XAmzContentSha256 *string                        `header:"style=simple,explode=false,name=X-Amz-Content-Sha256"`
	XAmzCredential    *string                        `header:"style=simple,explode=false,name=X-Amz-Credential"`
	XAmzDate          *string                        `header:"style=simple,explode=false,name=X-Amz-Date"`
	XAmzSecurityToken *string                        `header:"style=simple,explode=false,name=X-Amz-Security-Token"`
	XAmzSignature     *string                        `header:"style=simple,explode=false,name=X-Amz-Signature"`
	XAmzSignedHeaders *string                        `header:"style=simple,explode=false,name=X-Amz-SignedHeaders"`
	XAmzTarget        RegisterInstanceXAmzTargetEnum `header:"style=simple,explode=false,name=X-Amz-Target"`
}

type RegisterInstanceRequest struct {
	Headers RegisterInstanceHeaders
	Request shared.RegisterInstanceRequest `request:"mediaType=application/json"`
}

type RegisterInstanceResponse struct {
	ContentType               string
	RegisterInstanceResult    *shared.RegisterInstanceResult
	ResourceNotFoundException *interface{}
	StatusCode                int64
	ValidationException       *interface{}
}

File: SDKs/amazonaws.com/opsworks/2013-02-18/typescript/src/sdk/models/operations/registerinstance.ts
import { Metadata, SpeakeasyBase } from "../../../internal/utils/utils";
import * as shared from "../shared";

export enum RegisterInstanceXAmzTargetEnum {
    OpsWorks20130218RegisterInstance = "OpsWorks_20130218.RegisterInstance"
}


export class RegisterInstanceHeaders extends SpeakeasyBase {
  @Metadata({ data: "header, style=simple;explode=false;name=X-Amz-Algorithm" })
  xAmzAlgorithm?: string;

  @Metadata({ data: "header, style=simple;explode=false;name=X-Amz-Content-Sha256" })
  xAmzContentSha256?: string;

  @Metadata({ data: "header, style=simple;explode=false;name=X-Amz-Credential" })
  xAmzCredential?: string;

  @Metadata({ data: "header, style=simple;explode=false;name=X-Amz-Date" })
  xAmzDate?: string;

  @Metadata({ data: "header, style=simple;explode=false;name=X-Amz-Security-Token" })
  xAmzSecurityToken?: string;

  @Metadata({ data: "header, style=simple;explode=false;name=X-Amz-Signature" })
  xAmzSignature?: string;

  @Metadata({ data: "header, style=simple;explode=false;name=X-Amz-SignedHeaders" })
  xAmzSignedHeaders?: string;

  @Metadata({ data: "header, style=simple;explode=false;name=X-Amz-Target" })
  xAmzTarget: RegisterInstanceXAmzTargetEnum;
}


export class RegisterInstanceRequest extends SpeakeasyBase {
  @Metadata()
  headers: RegisterInstanceHeaders;

  @Metadata({ data: "request, media_type=application/json" })
  request: shared.RegisterInstanceRequest;
}


export class RegisterInstanceResponse extends SpeakeasyBase {
  @Metadata()
  contentType: string;

  @Metadata()
  registerInstanceResult?: shared.RegisterInstanceResult;

  @Metadata()
  resourceNotFoundException?: any;

  @Metadata()
  statusCode: number;

  @Metadata()
  validationException?: any;
}
